src/Widgets/PreferencesDialog.vala

```diff
@@ -19,6 +19,45 @@
 * Authored by: Felipe Escoto <felescoto95@hotmail.com>
 */
 
+/**
+ *  The purpose of this class is to easily create a new Switchbox for the
+ *  PreferencesDialog class.
+ */
+private class ENotes.PreferencesDialogSwitchBox {
+
+    private Gtk.Box container;
+    private Gtk.Switch switchbox;
+    private Gtk.Label  label;
+
+    public Gtk.Box SwitchBox { get { return this.container; } }
```

Comment:
You don't need to separate both the public and private variables like you did here :) (At least not in this case since it's a simple return)

But you can do:

`public Gtk.Box switch_box { get; private set; }`